run_iwyu_tests.py

```diff
@@ -42,6 +42,11 @@ def Partition(l, delimiter):
 
 def TestIwyuOnRelevantFiles(filename):
   logging.info('Testing iwyu on %s', filename)
+
+  skip = iwyu_test_util.CheckSkipConditions(filename)
```

Comment:
Shouldn't the skip-check happen in `TestIwyuOnRelativeFile` instead? I don't think this function is executed in the plain `./run_iwyu_tests.py` run mode.

Reply:
My mistake; it is. Still feels like it's better to parse/skip in `TestIwyuOnRelativeFile`, but I might be overlooking some complication.

Reply:
I had it here originally, moved it to `TestIwyuOnRelativeFile` and moved it back because `iwyu_test_utils.py` doesn't currently import `unit_tests`... so we can't throw the exception. Adding the import is trivial, but it's kinda nice that only `run_iwyu_tests.py` imports `unit_tests`. Happy to move if you prefer it that way.

Reply:
Aha, I see. Yeah, I think it's fine to import `unittest` in `iwyu_test_util.py` for this purpose.